cmake/Modules/VolkAddTest.cmake

```diff
@@ -164,10 +165,15 @@ function(VOLK_ADD_TEST test_name executable_name)
     #add the shell file as the test to execute;
     #use the form that allows for $<FOO:BAR> substitutions,
     #then combine the script arguments inside the script.
-    add_test(NAME qa_${test_name}
-      COMMAND ${SHELL} ${sh_file} ${TARGET_DIR_LIST}
-    )
-
+    if (NOT CMAKE_CROSSCOMPILING)
+        add_test(NAME qa_${test_name}
+            COMMAND ${SHELL} ${sh_file} ${TARGET_DIR_LIST}
+        )
+    else()
+        add_test(NAME qa_${test_name}
+            COMMAND ${SHELL} ${test_name}_test.sh ${TARGET_DIR_LIST}
+        )
+    endif(CMAKE_CROSSCOMPILING)
```

Comment:
So the difference here is `{sh_file}` vs `{test_name}_test.sh`. Wouldn't it be sensible to write the correct filename into the variable `{sh_file}` directly?

Reply:
sh_file contains the path to the shell file also. This path is based on the location of the build system, which is not where tests are installed on the target.

Reply:
Still, wouldn't it be better to write the value you need to `${sh_file}`? This would make this particular section simpler. The other section l142ff would benefit as well because comments on why things are done the way they are done could go there. I'd expect your short explanation why this is necessary there.